modules/jdbc/src/main/java/org/apache/ignite/internal/jdbc/JdbcResultSet.java

```diff
@@ -201,6 +205,37 @@ public JdbcResultSet(List<List<Object>> rows, List<JdbcColumnMeta> meta) throws
         initColumnOrder();
     }
 
+    JdbcResultSet getNextResultSet() throws SQLException {
+        try {
+            JdbcGetMoreResultsRequest req = new JdbcGetMoreResultsRequest(cursorId);
+            JdbcGetMoreResultsResult res = cursorHandler.getMoreResultsAsync(req).get();
+
+            if (!res.hasResults()) {
+                return null;
+            }
+
+            long cursorId0 = res.cursorId();
+
+            long updCount = 0;
+
+            if (res.isDml()) {
+                CompletableFuture<JdbcQueryFetchResult> fetch = cursorHandler.fetchAsync(
+                        new JdbcQueryFetchRequest(cursorId0, 1));
+                JdbcQueryFetchResult fetchRes = fetch.get();
+                updCount = (long) fetchRes.items().get(0).get(0);
+            }
+
+            return new JdbcResultSet(cursorHandler, stmt, cursorId0, getFetchSize(), false, Collections.emptyList(), res.isQuery(),
```

Comment:
I misunderstand here, if you suggest to call fetchAsync inside  of getMoreResultsAsync then we obtain blocking code, otherwise i miss your suggestion, can you show where i can obtain first page results plz ?

Reply:
oh, catch it, still in progress here

Reply:
but .. in case if some one not planned to use this prefetched results - we will make an extra work\send.

Reply:
well, you've done this already for DML (by the cost of extra round trip though)... And regarding just a query: why one might execute SELECT statement if they don't need a result? I believe the common case here is that every result will be consumed, thus prefetching the first page seems like a legit optimisation of the common case.

Reply:
Amen, i fix it )